daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/LedgerInteraction.scala

```diff
@@ -728,6 +773,38 @@ class JsonLedgerClient(
       })
     }
   }
+  override def queryContractKey(party: SParty, templateId: Identifier, key: SValue)(
+      implicit ec: ExecutionContext,
+      mat: Materializer) = {
+    val req = HttpRequest(
+      method = HttpMethods.POST,
+      uri = uri.withPath(uri.path./("v1")./("fetch")),
+      entity = HttpEntity(
+        ContentTypes.`application/json`,
+        JsonLedgerClient.FetchKeyArgs(templateId, key.toValue).toJson.prettyPrint),
+      headers = List(Authorization(OAuth2BearerToken(token.value)))
+    )
+    for {
+      () <- validateTokenParty(party, "queryContractKey")
```

Comment:
Patterns on LHS of `<-` always add a `withFilter` in `for`s.

```suggestion
      _ <- validateTokenParty(party, "queryContractKey")
```

Reply:
Thanks! I keep forgetting that.